### PR TITLE
Implement and fix some modules functions

### PR DESCRIPTION
**Pull request by bookmist**

### CODE REVIEW

File: vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp [hunk 1]
```diff
@@ -145,28 +145,28 @@ EXPORT(int, _sceKernelExitCallback) {
...
-EXPORT(SceInt32, _sceKernelGetCallbackInfo, SceUID callbackId, Ptr<SceKernelCallbackInfo> pInfo) {
+EXPORT(SceInt32, _sceKernelGetCallbackInfo, SceUID callbackId, SceKernelCallbackInfo *pInfo) {
     const CallbackPtr cb = lock_and_find(callbackId, host.kernel.callbacks, host.kernel.mutex);
 
     if (!cb)
         return RET_ERROR(SCE_KERNEL_ERROR_UNKNOWN_CALLBACK_ID);
 
-    SceKernelCallbackInfo *info = pInfo.get(host.mem);
-    if (!info)
+    //SceKernelCallbackInfo *info = pInfo.get(host.mem);
```
Comment (Zangetsu38) [inline vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp:154]: you can delete this line if is not used anymore

File: vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp [hunk 2]
```diff
@@ -145,28 +145,28 @@ EXPORT(int, _sceKernelExitCallback) {
...
-    info->pCommon = reinterpret_cast<void *>(cb->get_user_common_ptr().address());
+    pInfo->callbackId = callbackId;
+    strncpy(pInfo->name, cb->get_name().c_str(), KERNELOBJECT_MAX_NAME_LENGTH);
+    pInfo->name[KERNELOBJECT_MAX_NAME_LENGTH] = '\0';
+    pInfo->attr = 0;
+    pInfo->threadId = cb->get_owner_thread_id();
+    pInfo->callbackFunc = cb->get_callback_function(); //reinterpret_cast<SceKernelCallbackFunction *>(cb->get_callback_function().address());
+    pInfo->notifyId = cb->get_notifier_id();
+    pInfo->notifyArg = cb->get_notify_arg();
+    pInfo->pCommon = cb->get_user_common_ptr(); //reinterpret_cast<void *>(cb->get_user_common_ptr().address());
```
Comment (Zangetsu38) [inline vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp:169]: same for here, no request keep // if using other things 🤔

File: vita3k/modules/SceNgsUser/SceNgs.cpp [hunk 3]
```diff
@@ -700,8 +700,17 @@ EXPORT(int, sceNgsVoiceResume, SceNgsVoiceHandle handle) {
...
 }
 
-EXPORT(int, sceNgsVoiceSetFinishedCallback) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceNgsVoiceSetFinishedCallback, SceNgsVoiceHandle voice_handle, Ptr<ngs::ModuleCallback> callback, Ptr<void> user_data) {
+    if (host.cfg.current_config.disable_ngs) {
+        return 0;
+    }
+
+    ngs::Voice *voice = voice_handle.get(host.mem);
```
Comment (Zangetsu38) [inline vita3k/modules/SceNgsUser/SceNgs.cpp:708]:
```suggestion
    ngs::Voice *voice = voice_handle.get(host.mem);
    if (!voice) {
        return RET_ERROR(SCE_NGS_ERROR_INVALID_ARG);
    }
```

File: vita3k/modules/SceNgsUser/SceNgs.cpp [hunk 4]
```diff
@@ -700,8 +700,17 @@ EXPORT(int, sceNgsVoiceResume, SceNgsVoiceHandle handle) {
...
+
```
Comment (Zangetsu38) [inline vita3k/modules/SceNgsUser/SceNgs.cpp:709]:
```suggestion
    if (!voice) {
        return RET_ERROR(SCE_NGS_ERROR_INVALID_ARG);
    }
```

File: vita3k/modules/SceAvPlayer/SceAvPlayer.cpp [hunk 5]
```diff
@@ -258,11 +261,11 @@ EXPORT(int32_t, sceAvPlayerAddSource, SceUID player_handle, Ptr<const char> path
         }
         free(host.mem, buf);
         temp_file.close();
+        host.kernel.run_guest_function(player_info->file_manager.close_file.address(), { player_info->file_manager.user_data });
         if (fs::file_size(temp_file_path) != file_size) {
             LOG_ERROR("File is corrupted or incomplete: {}", temp_file_path.string());
-            return -1;
+            return RET_ERROR(-1);
```
Comment (Zangetsu38) [inline vita3k/modules/SceAvPlayer/SceAvPlayer.cpp:267]: ret_error here no have sense, RET_ERROR is just for priint enum error in log, but here log is already writed

**Review by Zangetsu38 — CHANGES_REQUESTED** [review]
No exactly what commit exactly, but your pr kill Catherine and other persona dancing, crash in boot
look for fix it before

**Comment by bookmist** [comment]
> No exactly what commit exactly, but your pr kill Catherine crash in boot look for fix it before

I excluded commit causing this crash.

**Review by Zangetsu38 — APPROVED** [review]

**Pull request merged**